raiden/utils/nursery.py

```diff
@@ -150,7 +156,10 @@ def __exit__(
             # clear the resources when exiting.
             atexit.unregister(self._free_resources)
 
-        return None
+        log.debug(f"Exiting {self._stop.get()}")
+        if self._stop.exception:
+            raise self._stop.exception
+        return self._stop.get()
```

Comment:
This has to be rewritten:
- Formatting the log message will raise, which is not what we want.
- The whole `if` is doing what `get` does, so it can go away.
- If `get` has anything that is not an exception, it should not be returned, since this method return type is `Optional[bool]` and the result is `Any`.

Reply:
- removed the unnecessary log message
- removed potential for hiding `_stop.exception` by only conditionally setting in the `__exit__` function.
- changed return value